File: другое/тинькофф.аисд/4/A.py
import sys

n = int(sys.stdin.readline())
arr = list(map(int, sys.stdin.readline().split()))
m = int(sys.stdin.readline())

prefixSum = []
prefixXor = []
prefixSum.append(arr[0])
prefixXor.append(arr[0])

for i in range(1, n):
    prefixSum.append(prefixSum[i - 1] + arr[i])
    prefixXor.append(prefixXor[i - 1] ^ arr[i])


def giveSum(l, r):
    if l == 1:
        return prefixSum[r - 1]
    else:
        return prefixSum[r - 1] - prefixSum[l - 2]


def giveXor(l, r):
    if l == 1:
        return prefixXor[r - 1]
    else:
        return prefixXor[r - 1] - prefixXor[l - 2]


ans = []

for i in range(m):
    q, l, r = map(int, sys.stdin.readline().split())
    if q == 1:
        ans.append(giveSum(l, r))
    else:
        ans.append(giveXor(l, r))

sys.stdout.write("\n".join(map(str, ans)))

File: другое/тинькофф.аисд/4/A.js
var readline = require("readline");
var rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false,
});
var i = 0;
var n = 0;
var m = -1;
var arr = [];
var query = [];
rl.on("line", function (line) {
    if (i === 0) {
        n = Number(line);
        i++;
    }
    else if (i === 1) {
        arr = line.split(" ").map(Number);
        i++;
    }
    else if (i === 2) {
        m = Number(line);
        i++;
    }
    else {
        query.push(line.split(" ").map(Number));
        m--;
    }
    if (m === 0) {
        final();
        rl.close();
    }
});
function final() {
    var prefixSum = [];
    var prefixXor = [];
    prefixSum[0] = arr[0];
    prefixXor[0] = arr[0];
    for (var i_1 = 1; i_1 < arr.length; i_1++) {
        prefixSum[i_1] = prefixSum[i_1 - 1] + arr[i_1];
        prefixXor[i_1] = prefixXor[i_1 - 1] ^ arr[i_1];
    }
    var giveXor = function (l, r) {
        if (l === 1) {
            return prefixXor[r - 1];
        }
        else {
            return prefixXor[r - 1] ^ prefixXor[l - 2];
        }
    };
    var giveSum = function (l, r) {
        if (l === 1) {
            return prefixSum[r - 1];
        }
        else {
            return prefixSum[r - 1] - prefixSum[l - 2];
        }
    };
    for (var _i = 0, query_1 = query; _i < query_1.length; _i++) {
        var _a = query_1[_i], q = _a[0], l = _a[1], r = _a[2];
        if (q === 1) {
            console.log(giveSum(l, r));
        }
        else {
            console.log(giveXor(l, r));
        }
    }
}
